ui/components-react/src/components-react/filter-builder/FilterBuilderRuleGroup.tsx

```diff
@@ -206,3 +209,28 @@ const PropertyFilterBuilderGroupOrRule = React.memo(
     );
   }
 );
+
+const useRulePropertyFocus = (
+  currentGroupItemsLength: number,
+  groupRef: React.RefObject<HTMLDivElement>
```

Comment:
Usually, I'd expect a hook to return a `React.Ref`. Might be fine in this case, since it is not exported.

Reply:
Now that you mentioned this maybe it makes sense to change `React.RefObject<HTMLDivElement>` into `HTMLDivElement | null`?

Reply:
applied @saskliutas suggestion

Reply:
I think the current solution might lead to some subtle issues where Ref value is changed w/o a re-render (hence useEffect logic is not run until the next render).